### ISSUE TITLE
Should escape " as %22

### ISSUE THREAD

**Issue by eddyp** [ryanavella/outlook-exe#1]
If the body of the email contains the character " (and probably others such as &), this leads to an error when Outlook is called and something like this needs to be done in the caller:

```
use outlook_exe;


// some sourece for body, such a file read
// let body_file: String = fs::read_to_string(&args[1]).unwrap().parse().unwrap();
let body: String = body_file.replace("\"", "%22");

outlook_exe::MessageBuilder::new()
    .with_recipient("noreply@example.org")
    .with_subject("Hello, World!")
    .with_body(body)
    .with_attachment("C:/tmp/file.txt")
    .spawn()
    .unwrap();
```

I initially thought of sending a patch, but I saw the `Cow` usage which made me rethink always doing the replace.

Maybe an escape flag with an associated enum (with EscapeAll/EscapeList(Vec<char>) is also needed, but would probably have to change the builder to enforce the calling of the escape on the first position so the escape is done on all fields, not only the body?

**Comment by ryanavella** [comment]
Funny, I didn't expect anyone to actually use this crate. I was teaching myself how to publish to crates.io. I suppose I should probably support this now. 😄 

It appears that newlines, tabs, alphanumeric ASCII characters, and the following special characters work as expected:
``!#$'()*+,-./:;<=>@[\]^_`{|}~``

These characters however need to be escaped:
`"%&?`

Α quick and not-so-thorough test with my Ελληνικά keyboard shows that Unicode codepoints appear to work just fine. 

What bothers me, is that I expected `std::process::Command` to handle the escape characters for me. (it clearly escapes them correctly for e.g. newlines) I'm just pushing strings via `Command::arg(arg)`, which works internally by pushing `arg.to_os_string()` onto a `Vec<OsString>` for storage. Nothing crazy here. Then when you call `Command::spawn()`, it internally calls [`make_command_line`](https://github.com/rust-lang/rust/blob/6318d24ad8440fa30428b405be1174478e9536e3/src/libstd/sys/windows/process.rs#L449), which appears to have some logic for handling double-quote escapes already. It isn't clear to me whether this is something `std::process::Command` should be handling, or if I need to escape the characters myself.

**Comment by eddyp** [comment]
> 
> 
> Funny, I didn't expect anyone to actually use this crate. I was teaching myself how to publish to crates.io. I suppose I should probably support this now. 😄

I had a small use case, I liked the 0.0.0 version :)

> These characters however need to be escaped:
> `"%&?`
> What bothers me, is that I expected `std::process::Command` to handle the escape characters for me. (it clearly escapes them correctly for e.g. newlines) I'm just pushing strings via 

Note that the working escaping format is HTML/URL style that's why I gave the `"` turns into `%22` sequence example, as it took me some time to realize that. Not sure if it has anything to do with HTML format of the email, but the default option for my emails is plain text, so I don't think it does.

BTW, if there are so very few characters to escape, then I don't think my "escape only these chars" options makes sense, it should probably be an "escape them for me" vs "I already escaped it". Also, I don't think it makes sense to bring in a crate dep just for escaping, unless the list expands drastically.

**Comment by ryanavella** [comment]
Yeah it definitely looks reminiscent of percent-encoding of HTML/URL's. What is strange is that only a few characters need escaping. My guess is that Microsoft created their own custom escape encoding.

I'd like to fix this on my end for 0.0.1. But since the body needs to be modified in order to percent-encode it, I think I should change the API from this:

```
pub fn with_body<S>(self, body: S) -> Self
where
    S: Into<Cow<'a, str>>
```
... to this:
```
pub fn with_body<S>(self, body: S) -> Self
where
    S: Into<String>
```
So that we can have owned `String`'s internally and modify them. Do you have any objections to the new API?

My plan is to test it on the full printable ASCII set as well as the first 10 blocks of the Basic Multilingual Plane. (so it should include most Latin, Greek, and Cyrillic characters)

**Comment by ryanavella** [comment]
@eddyp I think I have a fix. It is a little patchy, but as far as I can tell it gets the job done.

I did a not-so-thorough test by creating a message with the following body:

```
!"#$%&'()*+,-./0123456789:;<=>?
@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_`abcdefghijklmnopqrstuvwxyz{|}~
 ¡¢£¤¥¦§¨©ª«¬­®¯°±²³´µ¶·¸¹º»¼½¾¿
ÀÁÂÃÄÅÆÇÈÉÊËÌÍÎÏÐÑÒÓÔÕÖ×ØÙÚÛÜÝÞßàáâãäåæçèéêëìíîïðñòóôõö÷øùúûüýþÿ
ĀāĂăĄąĆćĈĉĊċČčĎďĐđĒēĔĕĖėĘęĚěĜĝĞğĠġĢģĤĥĦħĨĩĪīĬĭĮįİıĲĳĴĵĶķĸĹĺĻļĽľĿ
ŀŁłŃńŅņŇňŉŊŋŌōŎŏŐőŒœŔŕŖŗŘřŚśŜŝŞşŠšŢţŤťŦŧŨũŪūŬŭŮůŰűŲųŴŵŶŷŸŹźŻżŽžſ
ƀƁƂƃƄƅƆƇƈƉƊƋƌƍƎƏƐƑƒƓƔƕƖƗƘƙƚƛƜƝƞƟƠơƢƣƤƥƦƧƨƩƪƫƬƭƮƯưƱƲƳƴƵƶƷƸƹƺƻƼƽƾƿ
ǀǁǂǃǄǅǆǇǈǉǊǋǌǍǎǏǐǑǒǓǔǕǖǗǘǙǚǛǜǝǞǟǠǡǢǣǤǥǦǧǨǩǪǫǬǭǮǯǰǱǲǳǴǵǶǷǸǹǺǻǼǽǾǿ
ȀȁȂȃȄȅȆȇȈȉȊȋȌȍȎȏȐȑȒȓȔȕȖȗȘșȚțȜȝȞȟȠȡȢȣȤȥȦȧȨȩȪȫȬȭȮȯȰȱȲȳȴȵȶȷȸȹȺȻȼȽȾȿ
ɀɁɂɃɄɅɆɇɈɉɊɋɌɍɎɏɐɑɒɓɔɕɖɗɘəɚɛɜɝɞɟɠɡɢɣɤɥɦɧɨɩɪɫɬɭɮɯɰɱɲɳɴɵɶɷɸɹɺɻɼɽɾɿ
ʀʁʂʃʄʅʆʇʈʉʊʋʌʍʎʏʐʑʒʓʔʕʖʗʘʙʚʛʜʝʞʟʠʡʢʣʤʥʦʧʨʩʪʫʬʭʮʯʰʱʲʳʴʵʶʷʸʹʺʻʼʽʾʿ
ˀˁ˂˃˄˅ˆˇˈˉˊˋˌˍˎˏːˑ˒˓˔˕˖˗˘˙˚˛˜˝˞˟ˠˡˢˣˤ˥˦˧˨˩˪˫ˬ˭ˮ˯˰˱˲˳˴˵˶˷˸˹˺˻˼˽˾˿
̴̵̶̷̸̡̢̧̨̛̖̗̘̙̜̝̞̟̠̣̤̥̦̩̪̫̬̭̮̯̰̱̲̳̹̺̻̼̀́̂̃̄̅̆̇̈̉̊̋̌̍̎̏̐̑̒̓̔̽̾̿̕̚
͇͈͉͍͎̀́͂̓̈́͆͊͋͌ͅ͏͓͔͕͖͙͚͐͑͒͗͛ͣͤͥͦͧͨͩͪͫͬͭͮͯ͘͜͟͢͝͞͠͡ͰͱͲͳʹ͵Ͷͷͺͻͼͽ;Ϳ
΄΅Ά·ΈΉΊΌΎΏΐΑΒΓΔΕΖΗΘΙΚΛΜΝΞΟΠΡΣΤΥΦΧΨΩΪΫάέήίΰαβγδεζηθικλμνξο
πρςστυφχψωϊϋόύώϏϐϑϒϓϔϕϖϗϘϙϚϛϜϝϞϟϠϡϢϣϤϥϦϧϨϩϪϫϬϭϮϯϰϱϲϳϴϵ϶ϷϸϹϺϻϼϽϾϿ
ЀЁЂЃЄЅІЇЈЉЊЋЌЍЎЏАБВГДЕЖЗИЙКЛМНОПРСТУФХЦЧШЩЪЫЬЭЮЯабвгдежзийклмноп
рстуфхцчшщъыьэюяѐёђѓєѕіїјљњћќѝўџѠѡѢѣѤѥѦѧѨѩѪѫѬѭѮѯѰѱѲѳѴѵѶѷѸѹѺѻѼѽѾѿ
Ҁҁ҂҃҄҅҆҇҈҉ҊҋҌҍҎҏҐґҒғҔҕҖҗҘҙҚқҜҝҞҟҠҡҢңҤҥҦҧҨҩҪҫҬҭҮүҰұҲҳҴҵҶҷҸҹҺһҼҽҾҿ
ӀӁӂӃӄӅӆӇӈӉӊӋӌӍӎӏӐӑӒӓӔӕӖӗӘәӚӛӜӝӞӟӠӡӢӣӤӥӦӧӨөӪӫӬӭӮӯӰӱӲӳӴӵӶӷӸӹӺӻӼӽӾӿ
ԀԁԂԃԄԅԆԇԈԉԊԋԌԍԎԏԐԑԒԓԔԕԖԗԘԙԚԛԜԝԞԟԠԡԢԣԤԥԦԧԨԩԪԫԬԭԮԯ
```

I've pushed the change to the master branch. Before I publish version 0.0.1 to crates.io however, I just want to make sure you are okay with the change from `Into<Cow<'a, str>>` to `Into<String>`. It unfortunately makes heap allocations when they aren't strictly needed, but the code is a lot cleaner this way. Do you have any issues? I'm assuming it shouldn't affect your use-case.

**Comment by eddyp** [comment]
> I've pushed the change to the master branch. Before I publish version 0.0.1 to crates.io however, I just want to make sure you are okay with the change from `Into<Cow<'a, str>>` to `Into<String>`. It unfortunately makes heap allocations when they aren't strictly needed, but the code is a lot cleaner this way. Do you have any issues? I'm assuming it shouldn't affect your use-case.

No, no issues on my side.

**Comment by ryanavella** [comment]
Closed with 0.1.0
